src/features/filter.ts

```diff
@@ -1,12 +1,27 @@
-import { createSlice } from '@reduxjs/toolkit';
+import { PayloadAction, createSlice } from '@reduxjs/toolkit';
+import { Status } from '../types/Status';
 
-const initialState = {
+type State = {
+  query: string;
+  status: string;
+};
+
+const initialState: State = {
   query: '',
-  status: 'all',
+  status: Status.all,
 };
 
 export const filterSlice = createSlice({
   name: 'filter',
   initialState,
-  reducers: {},
+  reducers: {
+    status: (state: State, action: PayloadAction<string>) => ({
+      ...state,
+      status: action.payload,
+    }),
```

Comment:
```suggestion
    status: (state: State, action: PayloadAction<string>) => {
      state.status = action.payload,
    },
```
In the redux toolkit you can mutate the state